lms/djangoapps/courseware/tests/test_date_summary.py

```diff
@@ -340,21 +351,38 @@ def test_no_certificate_available_date_for_self_paced(self):
         self.assertFalse(block.is_enabled)
 
     @waffle.testutils.override_switch('certificates.instructor_paced_only', True)
+    def test_no_certificate_available_date_for_audit_course(self):
+        """
+        Tests that Certificate Available Date is not visible in the course "Important Course Dates" section
+        if the course only has audit mode.
+        """
+        course = create_course_run()
+        audit_user = self.create_user()
+
+        # Enroll learner in the audit mode and verify the course only has 1 mode (audit)
+        CourseEnrollmentFactory(course_id=course.id, user=audit_user, mode=CourseMode.AUDIT)
+        CourseMode.objects.get(course_id=course.id, mode_slug=CourseMode.VERIFIED).delete()
+        all_course_modes = CourseMode.modes_for_course(course.id)
+        self.assertEqual(len(all_course_modes), 1)
+        self.assertEqual(all_course_modes[0].slug, CourseMode.AUDIT)
+
+        course.certificate_available_date = datetime.now(utc) + timedelta(days=7)
+        self.enable_course_certificates(course)
```

Comment:
I don't think you need to add a certificate for this test, since the idea here is that it's an audit-only course with no certificate.

Reply:
@iloveagent57  Good catch, actually the `CertificateAvailableDate` object is only `is_enabled` when this is set (So I assume for the courses displaying Certificate Available Date must have this set). The background is, these courses are reruns and I assume, the certs information have been cloned but not the course modes.